pyquil/api/qpu.py

```diff
@@ -306,15 +290,69 @@ def wait_for_job(self, job_id, ping_time=None, status_time=None):
                             Defaults to the value specified in the constructor (2 seconds)
         :return: Completed Job
         """
-        def get_job_fn():
-            return self.get_job(job_id)
-        return wait_for_job(get_job_fn,
-                            ping_time if ping_time else self.ping_time,
-                            status_time if status_time else self.status_time)
+        return self._connection._wait_for_job(job_id, 'QPU', ping_time, status_time)
 
     def _wrap_program(self, program):
         return {
             "machine": "QPU",
             "program": program,
             "device": self.device_name
         }
+
+
+class QPU(QAM):
+    def __init__(self, connection: ForestConnection, device: Union[Device, str] = None):
```

Comment:
Instead of exposing ForestConnection - why not create a `ForestOptions` dataclass which contains the set of options? I think that could get rid of a lot of the weirdness/duplication that results from needed to share a constructor

Reply:
Let me think about this a bit